### PR TITLE
Implementing learning rate decay (a missing optional argument)

### PR DESCRIPTION
**Pull request by RikoNyberg**
Imitated the lr_decay implementation from [torch.optim.Adagrad](https://pytorch.org/docs/stable/_modules/torch/optim/adagrad.html#Adagrad)

### CODE REVIEW

**Comment by msaroufim** [comment]
Been 3 years since CLA check not signed, closing
